### ISSUE TITLE
Addresses are taken in wrong order from X-Forwarded-For header

### ISSUE THREAD

**Issue by JonasJasas** [dotnet/aspnetcore#19987]
### Describe the bug
Followed this post https://devblogs.microsoft.com/aspnet/forwarded-headers-middleware-updates-in-net-core-3-0-preview-6/ to add X-Forwarded-For header support for `HttpContext.Connection.RemoteIpAddress`. IP addresses inside X-Forwarded-For header value are taken from the right side when should be taken from the left. This way I am getting last proxy IP address instead of client IP.

### To Reproduce
- Follow this post https://devblogs.microsoft.com/aspnet/forwarded-headers-middleware-updates-in-net-core-3-0-preview-6/
- Get request using several HTTP proxies (to have multiple IPs in X-Forwarded-For).
- Use `HttpContext.Connection.RemoteIpAddress` to get client IP address.

### Further technical details
.NET Core SDK (reflecting any global.json):
 Version:   3.1.200
 Commit:    c5123d973b

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.04
 OS Platform: Linux
 RID:         ubuntu.18.04-x64
 Base Path:   /usr/share/dotnet/sdk/3.1.200/

Host (useful for support):
  Version: 3.1.2
  Commit:  916b5cba26

.NET Core SDKs installed:
  3.1.200 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

**Comment by Tratcher** [comment]
This is intentional, it's a question of trust. x-forwarded-for is easily spoofed so you should only consume values from trusted proxies. See ForwardLimit.
https://github.com/dotnet/aspnetcore/blob/7def10277abbb3086c4ee00e369faf6aa27d676a/src/Middleware/HttpOverrides/src/ForwardedHeadersOptions.cs#L58

**Comment by JonasJasas** [comment]
Proxy count can vary. Let's say I am using `App <=> Nginx <=> CloudFlare` but internal requests from other apps goes through Nginx. So external requests will be with different amount of IPs inside x-forwarded-for comparing to internal requests. What I need is just IP from the left side of x-forwarded-for header. It is standard feature in the other frameworks.

**Comment by Tratcher** [comment]
This implementation is pretty paranoid because too many people trust those IPs for authentication, georestrictions, etc.. Without some check on who set that value then it can be easily spoofed.

There are two trust checks here. One is ForwardLimit, how many proxies you are sitting behind. In the above example you could set this to 2 since you're less concerned about spoofing from internal requests.

The other trust check is KnownProxies/KnownNetworks. The next x-forwarded-for value will only be consumed if the current remote IP is a known proxy/network. That check happens again as you walk each step back along the x-fowarded-for chain.

You can turn both of these checks off by setting FowardLimit to null and clearing the KnownProxies/Networks, but we explicitly do not recommend it.

**Comment by cacodev** [comment]
I tried to use the forwarded middleware today to get the IP address of my client that is calling my web app and I ran into issues with the right to left parsing. Trying to get the client's IP address with at least one proxy. My company uses Pivotal Cloud Foundry (or VMWare Tanzu now I guess) and every time I push, my app gets proxied by a different GO router.  We have a large number of these and for me to add every IP address of every GO router would be painful. I would prefer to have a configuration item to allow left to right rather than the right to left.

Or maybe I need a way to specify a wildcard value for the proxy (ie allow any proxy with a "10.11.12.*" address)
